src/ClientRequestContext.h

```diff
@@ -82,6 +82,9 @@ class ClientRequestContext : public RefCountable
 #endif
     ErrorState *error = nullptr; ///< saved error page for centralized/delayed processing
     bool readNextRequest = false; ///< whether Squid should read after error handling
+#if FOLLOW_X_FORWARDED_FOR
+    uint32_t currentHopNumber = 0;
```

Comment:
This name is a bit too generic for a very XFF-specific value.
```suggestion
    uint32_t currentXffHopNumber = 0;
```

Reply:
I was going to polish this manually later, but if we are doing this polish via change requests, then I can suggest:

```suggestion
    size_t xffAddressesChecked = 0;
```

or, if you really insist,

```suggestion
    size_t currentXffHopNumber = 0;
```

Rationale: We should not be counting bits in this context, and this data member is counting the number of address checks performed rather than some hop position or "number" (e.g., if we skip an XFF address for some reason, the counter should not go up even though we would be looking at the next address after that skip).

Reply:
AIUI this is a recursion counter to limit the stack size for this old semi-async code. A full fix would store the unprocessed XFF values and step over them in async Calls.

FTR; this logic must never "skip" XFF addresses. Every value MUST meet the criteria of being a valid representation of a raw-IP accepted by the ACLs - or the whole process stops and uses the previously found IP.

Reply:
I disagree with some of the above assertions, but since they do not really contradict my code change recommendations, I will save time by ignoring those disagreements.

Reply:
I went for `currentXffHopNumber` since you both seemed agree